src/controllers/CategoryController.js

```diff
@@ -0,0 +1,40 @@
+import BaseAPIController from './BaseAPIController'
+import CategoryProvider from '../providers/CategoryProvider'
+import request from '../service/request'
+export class CategoryController extends BaseAPIController {
+
+    /* Controller for get all categories*/
+    getAllCategories = async (req, res, next) => {
+        try {
+            let manage_data = await CategoryProvider.setPathForGetCategories(req.headers, req.method, req.isMagento);
```

Comment:
there is one issue in this structure where you are sending req.ismagento

right now we only have shopify/magento. but we will add woocommerce, bigommerce etc also later. then again all the code needs to be changed.

so write code assuming there are multiple backend platforms